theano/tensor/tests/test_basic.py

```diff
@@ -5114,14 +5114,17 @@ def __init__(self, name, shared=tensor._shared, op=Reshape, mode=None,
         self.ignore_topo = ignore_topo
         super(T_reshape, self).__init__(name)
 
-    def function(self, inputs, outputs):
+    def function(self, inputs, outputs, ignore_empty=False):
         f = function(inputs, outputs, mode=self.mode)
         if self.mode is not None or theano.config.mode != "FAST_COMPILE":
             topo = f.maker.fgraph.toposort()
             topo_ = [node for node in topo if not isinstance(node.op,
                                                              self.ignore_topo)]
-            assert len(topo_) == 1, topo_
-            assert type(topo_[0].op) is self.op
+            if ignore_empty:
+                assert len(topo_) <= 1, topo_
```

Comment:
I would do keep the second assert in case it is of shape 1. Otherwise, I would assert the shape is 0 and would rename ignore_empty to empty.